commit 4fb93d589bc688fe9c59ebbb19c4eefaa6a98d2b
Author: Omar Sandoval <osandov@fb.com>
Date:   Wed Mar 1 10:42:38 2017 -0800

    loop: fix LO_FLAGS_PARTSCAN hang
    
    commit e02898b423802b1f3a3aaa7f16e896da069ba8f7 upstream.
    
    loop_reread_partitions() needs to do I/O, but we just froze the queue,
    so we end up waiting forever. This can easily be reproduced with losetup
    -P. Fix it by moving the reread to after we unfreeze the queue.
    
    Fixes: ecdd09597a57 ("block/loop: fix race between I/O and set_status")
    Reported-by: Tejun Heo <tj@kernel.org>
    Signed-off-by: Omar Sandoval <osandov@fb.com>
    Reviewed-by: Ming Lei <tom.leiming@gmail.com>
    Signed-off-by: Jens Axboe <axboe@fb.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linuxtest/dummy/rpi_1275_4fb93d589bc688fe9c59ebbb19c4eefaa6a98d2b.txt
===================================================================
--- /dev/null
+++ linuxtest/dummy/rpi_1275_4fb93d589bc688fe9c59ebbb19c4eefaa6a98d2b.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
